### PR TITLE
Observation groups and CareTeam Updates

### PR DESCRIPTION
**Pull request by ferret1964**
Fixes for supporting the CareTeam data fetch at OHSU and support for the new query /observationssegmented/?subject={id}&valueset={ValueSet} which will return each different code in a value set with a related date ordered collection of that code so that we can graph eGFR into different lines. I have added two new data points to the test data set to support this too.

Example: {{APIServer}}/observationssegmented/?subject=cc-pat-pnoelle&valueset=2.16.840.1.113762.1.4.1222.179.

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -94,6 +94,11 @@ details on Spring boot.
 | fhir.secure.server.address | FHIR_SECURE_SERVER_ADDRESS |Default secure fhir server | https://api.logicahealth.org/MCCeCarePlanTest/data | https://api.logicahealth.org/MCCeCarePlanTest/data |
 | logging.level.org.springframework | | | WARN | WARN |
 | mcc.patient.id.system | MCC_PATIENT_ID_SYSTEM | System used to determine the Patient Id| | | |
+| mcc.provider.pullbyrole | MCC_PROVIDER_PULLBYROLE | Allow provider contact info to be fecthed by practitioner role | TRUE | TRUE |
+| mcc.careteam.use.active | MCC_CARETEAM_USE_ACTIVE | Enable Contacts to look for careteams in none are a part of the selected careplan | TRUE | TRUE |
```
Comment (aschutzman-cognitive) [inline README.MD:98]: Change "in" to "if"

File: README.MD [hunk 2]
```diff
@@ -94,6 +94,11 @@ details on Spring boot.
...
+| mcc.careteam.use.careplan | MCC_CARETEAM_USE_CAREPLAN | Enable Contacts to us the currently selected Care plan to determine teams | TRUE | TRUE |
```
Comment (aschutzman-cognitive) [inline README.MD:99]: Change "us" to "use"

File: src/main/java/com/cognitive/nih/niddk/mccapi/controllers/ContactController.java [hunk 3]
```diff
@@ -26,12 +28,26 @@
     private final QueryManager queryManager;
     private final IR4Mapper mapper;
 
+    @Value("${mcc.careteam.use.active}")
+    private String useActiveCareplans;
+    boolean isUseActiveCareplans;
```
Comment (aschutzman-cognitive) [inline src/main/java/com/cognitive/nih/niddk/mccapi/controllers/ContactController.java:33]: Maybe change to "isUsingActiveCareplans"?

File: src/main/java/com/cognitive/nih/niddk/mccapi/controllers/ContactController.java [hunk 4]
```diff
@@ -113,7 +131,30 @@ public ContactController(QueryManager queryManager, IR4Mapper mapper) {
             }
         }
 
+        if (!foundCareTeam && isUseActiveCareplans)
+        {
+            //For some reason there was no careplan or associated caretem - So we will look for anyu active careteam
```
Comment (aschutzman-cognitive) [inline src/main/java/com/cognitive/nih/niddk/mccapi/controllers/ContactController.java:136]: "careteam"  "any"

File: src/main/java/com/cognitive/nih/niddk/mccapi/data/primative/MccCodeableConcept.java [hunk 5]
```diff
@@ -13,5 +15,48 @@
...
+                {
+                    fndCode = cd;
+                    break;
+                }
+            }
+        }
+        if (fndCode == null)
+        {
+            //Default to first code
+            fndCode=coding[0];
```
Comment (aschutzman-cognitive) [inline src/main/java/com/cognitive/nih/niddk/mccapi/data/primative/MccCodeableConcept.java:37]: Is there any chance the coding array will be empty?  I don't see where it is populated.

**Comment by aschutzman-cognitive** [comment]
Alright, added some comments, mostly just little typos.  When this is merged and pushed to docker I will work on adapting the provider/patient apps to accommodate the new eGFR valuesets.

File: src/main/java/com/cognitive/nih/niddk/mccapi/controllers/ContactController.java (hunk 3 above)
Comment (ferret1964) [reply src/main/java/com/cognitive/nih/niddk/mccapi/controllers/ContactController.java:33]: Should actually be useActiveCareTeams... will fix all over

**Pull request merged**